src/mediarithmics/plugins/ad-renderer/base/AdRendererBasePlugin.ts

```diff
@@ -101,6 +101,17 @@ export abstract class AdRendererBasePlugin<T extends AdRendererBaseInstanceConte
     instanceContext: T
   ): Promise<AdRendererPluginResponse>;
 
+  protected async getInstanceContext(creativeId: string, forceReload: boolean): Promise<T> {
+    if (!this.pluginCache.get(creativeId) || forceReload){
+      this.pluginCache.put(
+        creativeId,
+        this.instanceContextBuilder(creativeId, forceReload).catch(() => this.pluginCache.del(creativeId)),
```

Comment:
You need to re-throw the exception after the cache invalidation to propagate the error and not return a successful `undefined` (same for others).